File: Bookcase web/src/main/java/cz/vhromada/book/web/converter/CategoryConverter.java
package cz.vhromada.book.web.converter;

import cz.vhromada.book.entity.Category;

import ma.glasnost.orika.MappingContext;
import ma.glasnost.orika.converter.BidirectionalConverter;
import ma.glasnost.orika.metadata.Type;
import org.springframework.stereotype.Component;

/**
 * A class represents converter between {@link Category} and {@link Integer}.
 *
 * @author Vladimir Hromada
 */
@Component("categoryConverter")
public class CategoryConverter extends BidirectionalConverter<Category, Integer> {

    @Override
    public Integer convertTo(final Category source, final Type<Integer> type, final MappingContext mappingContext) {
        if (source == null) {
            return null;
        }

        return source.getId();
    }

    @Override
    public Category convertFrom(final Integer source, final Type<Category> type, final MappingContext mappingContext) {
        if (source == null) {
            return null;
        }

        final Category category = new Category();
        category.setId(source);

        return category;
    }

}

File: Bookcase core/src/main/kotlin/cz/vhromada/book/converter/CategoryConverter.kt
package cz.vhromada.book.converter

import cz.vhromada.book.entity.Category
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component

/**
 * A class represent converter for category.
 *
 * @author Vladimir Hromada
 */
@Component("categoryConverter")
class CategoryConverter : Converter<Category, cz.vhromada.book.domain.Category> {

    /**
     * Returns converted [cz.vhromada.book.domain.Category] from [Category].
     *
     * @param source source
     * @return converted category
     */
    override fun convert(source: Category): cz.vhromada.book.domain.Category {
        return cz.vhromada.book.domain.Category(source.id, source.name, source.position)
    }

}
